Add MSTest as supported test adapter (#40350)
dotnet/docs: docs/core/testing/unit-testing-platform-intro.md

`Microsoft.Testing.Platform` is open source. You can find `Microsoft.Testing.Platform` code in [microsoft/testfx](https://github.com/microsoft/testfx/tree/main/src/Platform/Microsoft.Testing.Platform) GitHub repository.

> # Supported test frameworks

Reviewer: ```suggestion
## Supported test frameworks
```

> * MSTest a.k.a. [MSTest runner](unit-testing-mstest-runner-intro.md)

Reviewer: I think the section is good. I would maybe changed the wording for MSTest runner.

Something like

MSTest. In MSTest, the support of Microsoft.Testing.Platform is done via MSTest runner.

(I'm on phone so I have kept the back quote or links)

Author: applied

> * MSTest. In MSTest, the support of `Microsoft.Testing.Platform` is done via [MSTest runner](unit-testing-mstest-runner-intro.md).

Reviewer: ```suggestion
* MSTest. In MSTest, the support of `Microsoft.Testing.Platform` is done via [MSTest runner](unit-testing-mstest-runner-intro.md).
* NUnit: work in progress - refer to https://github.com/nunit/nunit3-vs-adapter/issues/1152
* xUnit: work in progress - refer to https://github.com/xunit/visualstudio.xunit/issues/402
```

`Microsoft.Testing.Platform` is open source. You can find `Microsoft.Testing.Platform` code in [microsoft/testfx](https://github.com/microsoft/testfx/tree/main/src/Platform/Microsoft.Testing.Platform) GitHub repository.

## Supported test frameworks

* MSTest. In MSTest, the support of `Microsoft.Testing.Platform` is done via [MSTest runner](unit-testing-mstest-runner-intro.md).

> * NUnit: work in progress - refer to https://github.com/nunit/nunit3-vs-adapter/issues/1152

Reviewer 2: ```suggestion
* NUnit: work in progress, for more information, see [https://github.com/nunit/nunit3-vs-adapter/issues/1152](Microsoft Testing Platform for NUnit).
```

> * NUnit: work in progress, for more information, see [https://github.com/nunit/nunit3-vs-adapter/issues/1152](Microsoft Testing Platform for NUnit).

Reviewer: ```suggestion
* NUnit: work in progress, for more information, see [Microsoft Testing Platform for NUnit](https://github.com/nunit/nunit3-vs-adapter/issues/1152).
```

> * xUnit: work in progress - refer to https://github.com/xunit/visualstudio.xunit/issues/402

Reviewer 2: ```suggestion
* xUnit: work in progress, for more information, see [Microsoft Testing Platform for xUnit](https://github.com/xunit/visualstudio.xunit/issues/402).
```
